data/backend/authentication/views.py

```diff
@@ -0,0 +1,188 @@
+                    k: " ".join(v)
+                    for k, v in (form.errors.items() if form.errors else [])
+                }
+                return JsonResponse(
+                    {
+                        "success": False,
+                        "msg": _("Your form has errors: ")
+                        + " ".join([f"{k}: {v}" for k, v in errors.items()]),
+                        "errors": errors,
+                    }
+                )
+        else:
+            # registration in pure django (without elm)
+            form = RegistrationForm(request)
+            if form.is_valid():
+                form.save()
+                return redirect("registration-requested")
+            else:
+                return render(
+                    request,
+                    "registration/register.html",
+                    {
+                        "form": form,
+                        "site_header": "Ecobalyse",
+                        "site_title": "Ecobalyse",
+                        "title": _("Register"),
+                    },
+                )
+    else:
+        # registration form in pure django (without elm)
+        form = RegistrationForm()
+        return render(
+            request,
+            "registration/register.html",
+            {
+                "form": form,
+                "site_header": "Ecobalyse",
+                "site_title": "Ecobalyse",
+                "title": _("Register"),
+            },
+        )
+
+
+class LoginView(MailauthLoginView):
+    extra_context = {
+        "site_header": "Ecobalyse",
+        "site_title": "Ecobalyse",
+        "title": _("Login"),
+    }
+
+    def post(self, request, *a, **kw):
+        if request.path.endswith(".json/"):
+            form = EmailLoginForm(
+                request=request, data=json.loads(request.body.decode("utf-8"))
+            )
+            if form.is_valid():
+                form.save()
+                return JsonResponse(
+                    {
+                        "success": True,
+                        "msg": _("The link is valid for %s min")
+                        % (getattr(settings, "LOGIN_URL_TIMEOUT", 900) / 60),
+                    }
+                )
+            else:
+                return JsonResponse({"success": False, "msg": _("Invalid form data")})
+        else:
+            form = self.get_form()
+            if form.is_valid():
+                return self.form_valid(form)
+            else:
+                return self.form_invalid(form)
+
+
+class RegistrationRequestedView(generic.TemplateView):
+    """confirmation that the email has beed stored"""
+
+    template_name = "registration/registration_requested.html"
+
+
+class Activate(MailauthLoginTokenView):
+    """login and activate the disabled account"""
+
+    signer = signing.UserSigner()
+    max_age = getattr(settings, "LOGIN_URL_TIMEOUT", 60 * 15)
+    single_use = getattr(settings, "LOGIN_TOKEN_SINGLE_USE", True)
```

Comment:
What's the exact effect of this? It seems contradictory with using a persistent token for API authentication

Reply:
Replacing 900 with None so it is persistent